backends/rapidpro/status.go

```diff
@@ -281,6 +286,14 @@ UPDATE msgs_msg SET
 		ELSE 
 			next_attempt 
 		END,
+	failed_reason = CASE
+		WHEN
+			error_count >= 2
+		THEN
+			'E'
+		ELSE
+			failed_reason
+	    END,
```

Comment:
we have 3 different sql queries for updating message statuses and we don't do a great job of keeping them all in sync.. gonna look at reducing that to two queries